### PR TITLE
Add Java. Include base test examples and a few sample solutions

### PR DESCRIPTION
**Pull request by PhaseRush**
I've edited the `run.md` file to add a detailed instructions on how to get the project set up. Seeking comments regarding that.

### CODE REVIEW

**Comment by PhaseRush** [comment]
There are some other things left to do, such as configuring a linter and some small details here and there. But this is a good first effort and I'd like to push it before I lose steam.

**Review by dataprob — COMMENTED** [review]
Great work Requiem! I'll spend some time going thru the instructions to run

**Comment by dataprob** [comment]
Figured out how to do this in vscode, I'll add to doc later too.

File: doc/run.md [hunk 1]
```diff
@@ -33,12 +33,60 @@ To run all tests:
...
+Get IntelliJ [from JetBrains](https://www.jetbrains.com/idea/download). If you have a student email, you can get the Ultimate version for free. The community edition is perfectly adequate for this repo.
+
+(Recommended) - [Link IntelliJ with your GitHub](https://www.jetbrains.com/help/idea/github.html)
+
+(Required) - Make sure you have Java installed. This repo requires Java 14+; I'd recommend Java 17 or 18. You can install java on your system first, or [install it through IntelliJ](https://www.jetbrains.com/help/idea/sdk.html)
+
+(Recommended) - [Install gradle natively](https://gradle.org/install/). This is not required, and the wrapper this repo ships with should be enough to bootstrap any actions. However, installing gradle should make things a bit smoother in the beginning.
+
+---
+1. [Clone the repo via IntelliJ](https://www.jetbrains.com/help/idea/cloning-repository.html). ![cloning in intellij](figs/java/1_1_clone.png)
```
Comment (dataprob) [inline doc/run.md:65]: For all images in markdown, add a new line b4 and after, o.w. you have text and image on same line.
See how I did in GitCube https://github.com/LeetCube/GitCube/blob/main/pr/vsc.md?plain=1

**Review by dataprob — CHANGES_REQUESTED** [review]
See my comment about images in `run.md`, o.w. all good!

**Review by dataprob — APPROVED** [review]
Awesome!

**Pull request merged**